addons/mass_mailing/models/mass_mailing.py

```diff
@@ -433,9 +447,16 @@ def name_create(self, name):
     @api.multi
     def copy(self, default=None):
         self.ensure_one()
+        # Cleaning archived contact_list_ids
         default = dict(default or {},
-                       name=_('%s (copy)') % self.name)
-        return super(MassMailing, self).copy(default=default)
+                       name=_('%s (copy)') % self.name,
+                       contact_list_ids=self.contact_list_ids.ids)
+        res = super(MassMailing, self).copy(default=default)
+        # Re-evaluating the domain
+        body_html = res.body_html
+        res._onchange_model_and_list()
```

Comment:
Does this work the same way in this version?

Reply:
Yes. I just tested it on runbot and it works as expected